File: 0976-largest-perimeter-triangle/0976-largest-perimeter-triangle.cpp
class Solution {
public:
    int largestPerimeter(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int i=nums.size()-1;
        while(i>=2 && nums[i-2]+nums[i-1]<=nums[i])
            i--;
        return i>=2?nums[i-2]+nums[i-1]+nums[i]:0;
    }
};

File: 0976-largest-perimeter-triangle/0976-largest-perimeter-triangle.java
class Solution {
    public int largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        for (int i = n - 1; i >= 2; i--) {
            for (int j = i - 1; j >= 1; j--) {
                int k = j - 1;
                while (k >= 0 && nums[i] < nums[j] + nums[k] && nums[k] != 0)
                    return nums[i] + nums[j] + nums[k--];
            }
        }
        return 0;
    }
}